Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs

```diff
@@ -607,145 +616,64 @@ private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
 
             if (!SpawnMap(uid, nukeops))
             {
-                Logger.InfoS("nukies", "Failed to load map for nukeops");
+                _sawmill.Info("Failed to load map for nukeops");
                 continue;
             }
 
-            // Basically copied verbatim from traitor code
-            var playersPerOperative = nukeops.PlayersPerOperative;
-            var maxOperatives = nukeops.MaxOps;
+            //Handle there being nobody readied up
+            if (ev.PlayerPool.Count == 0)
+                continue;
 
-            // Dear lord what is happening HERE.
-            var everyone = new List<ICommonSession>(ev.PlayerPool);
-            var prefList = new List<ICommonSession>();
-            var medPrefList = new List<ICommonSession>();
-            var cmdrPrefList = new List<ICommonSession>();
-            var operatives = new List<ICommonSession>();
+            var commanderEligible = _antagSelection.GetEligibleSessions(ev.PlayerPool, nukeops.CommanderRoleProto, false);
+            var agentEligible = _antagSelection.GetEligibleSessions(ev.PlayerPool, nukeops.MedicRoleProto, false);
+            var operativeEligible = _antagSelection.GetEligibleSessions(ev.PlayerPool, nukeops.OperativeRoleProto, false);
 
-            // The LINQ expression ReSharper keeps suggesting is completely unintelligible so I'm disabling it
-            // ReSharper disable once ForeachCanBeConvertedToQueryUsingAnotherGetEnumerator
-            foreach (var player in everyone)
-            {
-                if (!ev.Profiles.ContainsKey(player.UserId))
-                {
-                    continue;
-                }
+            var nukiesToSelect = 6;//_antagSelection.CalculateAntagNumber(_playerManager.PlayerCount, nukeops.PlayersPerOperative, nukeops.MaxOps);
```

Comment:
fix this

Reply:
The shame, forgot to remove it after testing.